docs(summary-detail): add website documentation (#3760)
twilio-labs/paste: packages/paste-website/src/pages/components/summary-detail/index.mdx

  title="Standard Summary Detail"
  storyID="components-summarydetail--default"
  height="200px"
  code={`
<SummaryDetail>
  <SummaryDetailHeading>
    <SummaryDetailToggleButton aria-label="BOOP" />
    <SummaryDetailHeadingContent>
      <Text as="p" fontWeight="fontWeightBold">
        Inbound Call
      </Text>
      <Text as="p" fontSize="fontSize20" fontWeight="fontWeightLight">
        Jan 11, 2023
      </Text>
    </SummaryDetailHeadingContent>
  </SummaryDetailHeading>
  <SummaryDetailContent>
    <Text as="strong">Agent: </Text>
    <Text as="span">John Doe</Text>
    <Text as="span" marginX="space40" color="colorTextWeaker">
      |
    </Text>
    <Text as="strong">Duration: </Text>
    <Text as="span">3m 14s</Text>
  </SummaryDetailContent>
</SummaryDetail>`}
/>

## Guidelines

### About Summary Detail

The Summary Detail component displays a summary of information that can be expanded to show more information.
It can be used when you need more click targets & actions in a Disclosure Heading.

### Accessibility

- Only the toggle button is tabbable and clickable to reveal the content.
- The toggle button is labelled by the `SummaryDetailHeadingContent` component.

>   - If the heading content is unclear too verbose, you can provide an `aria-label` to `SummaryDetailToggleButton` directly to enable a more descriptive label.

Reviewer: ```suggestion
  - If the heading content is unclear or too verbose, you can provide an `aria-label` to `SummaryDetailToggleButton` directly to enable a more descriptive label.
```

        storybookUrl: '/?path=/story/components-summarydetail--default',
      },
    },
  };
};

<StoryPreview
  title="Standard Summary Detail"
  storyID="components-summarydetail--default"
  height="200px"
  code={`
<SummaryDetail>
  <SummaryDetailHeading>
    <SummaryDetailToggleButton aria-label="BOOP" />
    <SummaryDetailHeadingContent>
      <Text as="p" fontWeight="fontWeightBold">
        Inbound Call
      </Text>
      <Text as="p" fontSize="fontSize20" fontWeight="fontWeightLight">
        Jan 11, 2023
        <Anchor href="#" marginLeft="space30">
          (View transcript)
        </Anchor>
      </Text>
    </SummaryDetailHeadingContent>
  </SummaryDetailHeading>
  <SummaryDetailContent>
    <Text as="strong">Agent: </Text>
    <Text as="span">John Doe</Text>
    <Text as="span" marginX="space40" color="colorTextWeaker">
      |
    </Text>
    <Text as="strong">Duration: </Text>
    <Text as="span">3m 14s</Text>
  </SummaryDetailContent>
</SummaryDetail>`}
/>

## Guidelines

### About Summary Detail

The Summary Detail component displays a summary of information that can be expanded to show more information.

> It can be used when you need more click targets & actions in a Disclosure Heading.

Reviewer 2: ```suggestion
It can be used when you need more click targets and actions in a Disclosure Heading.
```

### Accessibility

- Only the toggle button is tabbable and clickable to reveal the content.
- The toggle button is labelled by the `SummaryDetailHeadingContent` component.
  - If the heading content is unclear or too verbose, you can provide an `aria-label` to `SummaryDetailToggleButton` directly to enable a more descriptive label.

## Examples

### Basic usage

> This example shows how you can compose the [Text component](/primitives/text) to create a
> simple and clean looking Summary Detail.

Reviewer 2: ```suggestion
This example shows how you can use the [Text component](/primitives/text) to compose a
basic Summary Detail.
```

<StoryPreview
  title="Standard Summary Detail"
  storyID="components-summarydetail--default"
  height="200px"
  code={`
<SummaryDetail>
  <SummaryDetailHeading>
    <SummaryDetailToggleButton aria-label="BOOP" />
    <SummaryDetailHeadingContent>
      <Text as="p" fontWeight="fontWeightBold">
        Inbound Call
      </Text>
      <Text as="p" fontSize="fontSize20" fontWeight="fontWeightLight">
        Jan 11, 2023
        <Anchor href="#" marginLeft="space30">
          (View transcript)
        </Anchor>
      </Text>
    </SummaryDetailHeadingContent>
  </SummaryDetailHeading>
  <SummaryDetailContent>
    <Text as="strong">Agent: </Text>
    <Text as="span">John Doe</Text>
    <Text as="span" marginX="space40" color="colorTextWeaker">
      |
    </Text>
    <Text as="strong">Duration: </Text>
    <Text as="span">3m 14s</Text>
  </SummaryDetailContent>
</SummaryDetail>`}
/>

### Open by default

> You can set the content to be visible by default.

Reviewer 2: ```suggestion
You can set the content to be visible by default with the `visible` prop.
```
